### PR TITLE
Cluster deletion refactoring

### PR DESCRIPTION
**Pull request by m1kola**
Just addresses minor things from https://github.com/Azure/ARO-RP/pull/1541#pullrequestreview-697020905

@bryanro92 please take a look.

### CODE REVIEW

**Comment by m1kola** [comment]
Puting into draft. First need to deal with https://github.com/Azure/ARO-RP/pull/1568 which will cause merge conflicts here.

**Review by bryanro92 — DISMISSED** [review]
thanks. makes sense. cleaner approach.

File: pkg/cluster/delete.go [hunk 1]
```diff
@@ -272,8 +272,36 @@ func (m *manager) Delete(ctx context.Context) error {
...
+		err = nil
+	}
+	if azureerrors.HasAuthorizationFailedError(err) {
+		err = nil
+	}
+	if err != nil {
+		return err
+	}
+
+	return nil
```
Comment (nilsanderselde) [inline pkg/cluster/delete.go:299]:
```suggestion
	return err
```

**Comment by m1kola** [comment]
Rebased on top of the master to fix conflicts. Also addressed PR feedback.

**Review by mjudeikis — APPROVED** [review]

**Pull request merged**